haskell.nix

```diff
@@ -16,23 +16,26 @@ in self-hs: super-hs:
     sbv_prepatch = pkgs.haskell.lib.dontCheck (self-hs.callCabal2nix "sbv"
       (builtins.fetchGit {
         url = "https://github.com/LeventErkok/sbv";
-        rev = "59c1cf37070a5423113f950bf071804257756c7a";
-      }) 
+        rev = "b64905e2698c320ac14ffbad53325d33081839fb";
+      })
       {inherit (pkgs) z3;});
 
   in {
     restless-git = dontCheck "restless-git" (./src/restless-git);
     wreq = pkgs.haskell.lib.doJailbreak super-hs.wreq;
 
-    # we use a pretty bleeding edge sbv version
     sbv = sbv_prepatch.overrideAttrs (attrs: {
-      postPatch = ''
-      sed -i -e 's|"z3"|"${pkgs.z3}/bin/z3"|' Data/SBV/Provers/Z3.hs
-      sed -i -e 's|"cvc4"|"${pkgs.cvc4}/bin/cvc4"|' Data/SBV/Provers/CVC4.hs'';
+      postPatch =
+        if wrapped
```

Comment:
You could use `lib.optionalString wrapped ...` here instead. Cf https://nixos.org/manual/nixpkgs/stable/#function-library-lib.strings.optionalString